fathom/src/surface.rs

```diff
@@ -164,11 +162,101 @@ impl<'arena> Term<'arena, ByteRange> {
         interner: &RefCell<StringInterner>,
         scope: &'arena Scope<'arena>,
         source: &'source str,
-    ) -> Result<Term<'arena, ByteRange>, ParseError<'source>> {
-        grammar::TermParser::new().parse(interner, scope, lexer::tokens(source))
+    ) -> Result<Term<'arena, ByteRange>, ParseMessage> {
+        use lalrpop_util::ParseError;
+
+        grammar::TermParser::new()
+            .parse(interner, scope, lexer::tokens(source))
+            .map_err(|err| match err {
+                ParseError::InvalidToken { location } => ParseMessage::InvalidToken {
+                    range: ByteRange::new(location, location),
+                },
+                ParseError::UnrecognizedEOF { location, expected } => {
+                    ParseMessage::UnrecognizedEof {
+                        range: ByteRange::new(location, location),
+                        expected, // TODO: convert to descriptions?
+                    }
+                }
+                ParseError::UnrecognizedToken {
+                    token: (start, token, end),
+                    expected,
+                } => ParseMessage::UnrecognizedToken {
+                    range: ByteRange::new(start, end),
+                    token: token.description(),
+                    expected,
+                },
+                ParseError::ExtraToken {
+                    token: (start, token, end),
+                } => ParseMessage::ExtraToken {
+                    range: ByteRange::new(start, end),
+                    token: token.description(),
+                },
+                ParseError::User { error: () } => unreachable!(),
```

Comment:
I think syntax errors hit this branch. E.g. a fathom file with just `/` in it yields:

```
thread 'main' panicked at 'internal error: entered unreachable code', fathom/src/surface.rs:195:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

bug: compiler panicked at 'internal error: entered unreachable code'
 = panicked at: fathom/src/surface.rs:195:21
 = please file a bug report at: https://github.com/yeslogic/fathom/issues/new
```

Reply:
oh nooo haha. this is strange behavior from LALRPOP 🤔

Reply:
Oh wait, it might be because of:

https://github.com/yeslogic/fathom/blob/8e6703000788c7205efadf1c05ebdd3a88f29525/fathom/src/surface/lexer.rs#L65-L74

and: 

https://github.com/yeslogic/fathom/blob/8e6703000788c7205efadf1c05ebdd3a88f29525/fathom/src/surface.rs#L161-L169

Reply:
Should be addressed in https://github.com/yeslogic/fathom/pull/311/commits/f7c1f81428c5ebb69a794e0c12b295222fcdbcfc

Reply:
Confirming I get an error now